### PR TITLE
Add rules tests

### PR DESCRIPTION
**Pull request by CohenIdo**

### CODE REVIEW

File: tests/product/tests/test_rules.py [hunk 1]
```diff
@@ -2,45 +2,137 @@
...
     k8s.start_agent(yaml_file=file_path, namespace=cloudbeat_agent.namespace)
+    # config_node_pre_test(k8s_client=k8s, api_client=api_client)
     time.sleep(5)
+
     yield k8s, api_client, cloudbeat_agent
     k8s_yaml_list = get_k8s_yaml_objects(file_path=file_path)
     k8s.stop_agent(yaml_objects_list=k8s_yaml_list)
 
 
+@pytest.fixture(scope='module')
```
Comment (CohenIdo) [inline tests/product/tests/test_rules.py:30]: create test dependencies

File: tests/product/tests/test_rules.py [hunk 2]
```diff
@@ -2,45 +2,136 @@
...
+                continue
+            findings = log.get('result').get('findings')
+            log_timestamp = datetime.datetime.strptime(log["time"], '%Y-%m-%dT%H:%M:%Sz')
+            if (log_timestamp - exec_timestamp).total_seconds() < 0:
+                continue
+
+            if findings:
+                for finding in findings:
+                    if rule_tag in finding.get('rule').get('tags'):
+                        iteration += 1
```
Comment (CohenIdo) [inline tests/product/tests/test_rules.py:93]: change log counter

File: tests/product/tests/test_rules.py [hunk 3]
```diff
@@ -2,45 +2,136 @@
...
+                        agent_evaluation = finding.get('result').get('evaluation')
+                        if agent_evaluation == expected:
+                            print(f"{iteration}: expected:"
+                                  f"{expected} tags:"
+                                  f"{finding.get('rule').get('tags')}, "
+                                  f"evidence: {finding.get('result').get('evidence')} ",
```
Comment (CohenIdo) [inline tests/product/tests/test_rules.py:99]: add log evidence

File: tests/product/tests/test_rules.py [hunk 4]
```diff
@@ -2,45 +2,136 @@
...
+                                                         namespace=namespace,
+                                                         since_seconds=2))
+        except:
+            continue
+        for log in logs:
+            if not log.get('result'):
+                continue
+            findings = log.get('result').get('findings')
+            log_timestamp = datetime.datetime.strptime(log["time"], '%Y-%m-%dT%H:%M:%Sz')
+            if (log_timestamp - exec_timestamp).total_seconds() < 0:
```
Comment (CohenIdo) [inline tests/product/tests/test_rules.py:87]: check logs that created after the command execution

File: tests/deploy/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml [hunk 5]
```diff
@@ -24,13 +24,34 @@ spec:
       - name: cloudbeat-test-pv-storage
         mountPath: /usr/src/app/tests/report
       - name: etc-kubernetes
```
Comment (CohenIdo) [inline tests/deploy/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml:26]: WIP

File: tests/deploy/values/ci.yml [hunk 6]
```diff
@@ -2,4 +2,5 @@ cloudbeat:
   deploy: true
 
 testData:
-  marker: ci_cloudbeat
```
Comment (CohenIdo) [inline tests/deploy/values/ci.yml:5]: WIP

File: tests/commonlib/io_utils.py [hunk 7]
```diff
@@ -25,7 +25,15 @@ def get_logs_from_stream(stream: str) -> list[Munch]:
...
         if log and "bundles" in log:
-            result.append(Munch(json.loads(log)))
+            try:
+                result.append(Munch(json.loads(log)))
+            except:
+                try:
+                    result.append(Munch(json.loads(log.replace("'", '"'))))
+                except Exception as e:
+                    print(e)
+                    continue
```
Comment (oren-zohar) [inline tests/commonlib/io_utils.py:35]: too broad of an exception, replace it with the expected one.

File: tests/deploy/values/ci.yml [hunk 8]
```diff
@@ -2,4 +2,5 @@ cloudbeat:
...
\ No newline at end of file
+#  marker: ci_cloudbeat
```
Comment (oren-zohar) [inline tests/deploy/values/ci.yml:5]: remove

File: tests/product/tests/test_rules.py [hunk 9]
```diff
@@ -2,45 +2,136 @@
...
 This module verifies correctness of retrieved findings by manipulating audit and remediation actions
 """
+import datetime
 import time
 
 import pytest
 from commonlib.io_utils import get_logs_from_stream, get_k8s_yaml_objects
 from pathlib import Path
-
+from product.tests.tests.file_system.file_system_test_cases import *
```
Comment (oren-zohar) [inline tests/product/tests/test_rules.py:11]: is there a better way to import it?

File: tests/product/tests/test_rules.py [hunk 10]
```diff
@@ -2,45 +2,136 @@
...
         k8s.stop_agent(get_k8s_yaml_objects(file_path=file_path))
     k8s.start_agent(yaml_file=file_path, namespace=cloudbeat_agent.namespace)
     time.sleep(5)
+
     yield k8s, api_client, cloudbeat_agent
     k8s_yaml_list = get_k8s_yaml_objects(file_path=file_path)
     k8s.stop_agent(yaml_objects_list=k8s_yaml_list)
 
 
+@pytest.fixture(scope='module')
```
Comment (oren-zohar) [inline tests/product/tests/test_rules.py:29]: can be moved to conftest?

File: tests/product/tests/test_rules.py [hunk 11]
```diff
@@ -2,45 +2,136 @@
...
+
 @pytest.mark.rules
 @pytest.mark.parametrize(
     ("rule_tag", "command", "param_value", "resource", "expected"),
     [
-        ('CIS 1.1.1', 'chmod', '700', '/etc/kubernetes/manifests/kube-apiserver.yaml', 'failed'),
-        ('CIS 1.1.1', 'chmod', '644', '/etc/kubernetes/manifests/kube-apiserver.yaml', 'passed'),
-        ('CIS 1.1.2', 'chown', 'daemon:daemon', '/etc/kubernetes/manifests/kube-apiserver.yaml', 'failed'),
-        ('CIS 1.1.2', 'chown', 'root:root', '/etc/kubernetes/manifests/kube-apiserver.yaml', 'passed')
+        *cis_1_1_1,
```
Comment (oren-zohar) [inline tests/product/tests/test_rules.py:111]: spacing seems off

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: tests/product/tests/test_rules.py (hunk 10 above)
Comment (CohenIdo) [reply tests/product/tests/test_rules.py]: done

File: tests/product/tests/test_rules.py (hunk 9 above)
Comment (CohenIdo) [reply tests/product/tests/test_rules.py:10]: for now, now, let's have a look at it again after  the changes in your PR

File: tests/commonlib/io_utils.py (hunk 7 above)
Comment (CohenIdo) [reply tests/commonlib/io_utils.py]: done:)

File: tests/product/tests/conftest.py [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    # if 'etcd' not in users:
+    #     api_client.exec_command(container_name=node.metadata.name,
+    #                             command='useradd',
+    #                             param_value='-g etcd etcd',
+    #                             resource='')
+
+    # create stub file
+    # etcd_content = api_client.exec_command(container_name=node.metadata.name, command='ls',
+    #                                        param_value='/var/lib/etcd/', resource='')
+    # if 'some_file.txt' not in etcd_content:
```
Comment (oren-zohar) [inline tests/product/tests/conftest.py:48]: remove from PR

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**